src/librustc_codegen_ironox/intrinsic.rs

```diff
@@ -29,6 +30,9 @@ impl IntrinsicCallMethods<'tcx> for Builder<'a, 'll, 'tcx> {
     }
 
     fn expect(&mut self, cond: Value, expected: bool) -> Value {
-        unimplemented!("expect");
+        let expected_val = {
```

Comment:
Unnecessary curly brackets?

Reply:
Both `const_bool`, and `icmp` borrow `self` mutably, so I added that inner scope to make the borrow checker happy.

Reply:
Ah!